resourcemanager/pool/spool/spool.go

```diff
@@ -142,7 +145,12 @@ func (p *Pool) RunWithConcurrency(fns chan func(), concurrency uint32) error {
 
 // checkAndAddRunning is to check if a task can run. If can, add the running number.
 func (p *Pool) checkAndAddRunning(concurrency uint32) (conc int32, run bool) {
+	p.waiting.Add(1)
+	defer p.waiting.Add(-1)
```

Comment:
Move the waiting variable modification into Run? Because it may call `p.wg.Add(1)` after this checking and the check(`for p.waiting.Load() > 0 {...}`) in `ReleaseAndWait` may pass after `checkAndAddRunning` and before `p.wg.Add(1)`, then there is still possible race.

Reply:
You are right.